### PR TITLE
CRDCDH-1517 Add additional tooltips to Data Submission Dashboard

### PR DESCRIPTION
**Pull request by Alejandro-Vega**
### Overview

Added additional tooltips to the Validation control options as well as workflow buttons.

### Change Details (Specifics)

- Created a config file for tooltip static content
- Refactored the entire submit button logic to be more readable, invert logic so that it decides when to enable button instead of disable, and made it easier to identify which error is causing the button to remain disabled
- Created config file for submit button conditions. This is used to easily list out all the conditions for the submit button to remain disabled, as well as reasons for an admin override
 - Added `required` property to ensure some conditions won't be bypassed by the admin override 
 - Added ability to provide a tooltip for each of the conditions
 - Assigned tooltips for each of the conditions per requirements
 
**Submit button logic:**
---

**Enable submit button conditions: (NOTE: 'Initialized' = not null)**
- Validation should not currently be running (**REQUIRED**)
- Metadata and Data File should not have 'New' status (**REQUIRED**)
- Submission should not have submission level errors, such as orphaned files (**REQUIRED**)
- Metadata validation should be initialized for 'Delete' intention
- Metadata validation should be initialized for 'Metadata Only' submissions
- Data file validation should be initialized for 'Metadata and Data Files' submissions
- Metadata validation should be initialized for 'Metadata and Data Files' submissions
- There should be no validation errors for metadata or data files

**Admin Override submit button conditions: (It cannot override required conditions)**
- Submission has validation errors
- Submission is missing either metadata or data files for 'Metadata and Data Files' submissions
---

### Related Ticket(s)

[CRDCDH-1517](https://tracker.nci.nih.gov/browse/CRDCDH-1517)

### CODE REVIEW

File: src/utils/dataSubmissionUtils.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { ADMIN_OVERRIDE_CONDITIONS, SUBMIT_BUTTON_CONDITIONS } from "../config/SubmitButtonConfig";
 import { safeParse } from "./jsonUtils";
 
 export type SubmitInfo = {
```
Comment (amattu2) [inline src/utils/dataSubmissionUtils.ts:4]: TBD: The comment isn't appearing

**Review by amattu2 — CHANGES_REQUESTED** [review]
Awesome job with the submit action button refactoring. Extracting the ruleset to a config file will make it much easier in the future to add/edit rules.

I annotated a few suggestions, none of which are critical or show stoppers.

File: src/content/dataSubmissions/DataSubmission.tsx [hunk 2]
```diff
@@ -158,6 +158,8 @@ const submissionLockedStatuses: SubmissionStatus[] = [
   "Archived",
 ];
 
+const canSubmitRoles: User["role"][] = ["Submitter", "Organization Owner", "Data Curator", "Admin"];
```
[the remark this answers is not in the document]
Comment (Alejandro-Vega) [reply src/content/dataSubmissions/DataSubmission.tsx]: Agreed, thanks for catching this. I had made these changes before and forgot to return and update it. Updated in [5412c69](https://github.com/CBIIT/crdc-datahub-ui/pull/464/commits/5412c69eeba342eb64cec3bc56c4c3b00a216f53)

File: src/utils/dataSubmissionUtils.ts [hunk 3]
```diff
@@ -6,46 +7,115 @@ export type SubmitInfo = {
...
+ * @param {User["role"]} userRole - The role of the user (e.g., Admin, Submitter).
+ * @returns {SubmitButtonResult} - Returns an object indicating whether the submit button is enabled,
+ * whether the admin override is in effect, and an optional tooltip explaining why it is disabled.
  */
-export const shouldDisableSubmit = (submission: Submission, userRole: User["role"]): SubmitInfo => {
-  if (!userRole) {
-    return { disable: true, isAdminOverride: false };
+export const shouldEnableSubmit = (
+  submission: Submission,
+  userRole: User["role"]
```
[the remark this answers is not in the document]
Comment (Alejandro-Vega) [reply src/utils/dataSubmissionUtils.ts]: Updated in [cdacc5e](https://github.com/CBIIT/crdc-datahub-ui/pull/464/commits/cdacc5eaea6c4a50d27cbcc48a439922c81346c4).

File: src/components/DataSubmissions/ValidationControls.tsx [hunk 4]
```diff
@@ -276,24 +278,45 @@ const ValidationControls: FC = () => {
...
-                label="Validate Data Files"
-                disabled={!canValidateFiles}
-              />
-              <StyledRadioControl
-                value="All"
-                control={<StyledRadioButton readOnly={false} />}
-                label="Both"
-                disabled={!canValidateFiles || !canValidateMetadata}
-              />
+              <Tooltip
```
[the remark this answers is not in the document]
Comment (Alejandro-Vega) [reply src/components/DataSubmissions/ValidationControls.tsx]: Gap spacing reduced in [7832094](https://github.com/CBIIT/crdc-datahub-ui/pull/464/commits/78320949b2df6a9acfe54a44929db2fb9d97cf1d), LMK if this looks good.

File: src/utils/dataSubmissionUtils.ts (hunk 1 above)
Comment (Alejandro-Vega) [reply src/utils/dataSubmissionUtils.ts]: Thanks for catching this! Removed in [4857cf1](https://github.com/CBIIT/crdc-datahub-ui/pull/464/commits/4857cf1f9c987a538758c2e37cd5362d8165e930).

File: src/components/DataSubmissions/ValidationControls.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (amattu2) [reply src/components/DataSubmissions/ValidationControls.tsx]: Looks great! Thanks

**Review by amattu2 — APPROVED** [review]
LGTM! Nice job

**Pull request merged**